src/hal/src/lib.rs

```diff
@@ -469,7 +488,64 @@ pub struct Limits {
 
     /// The alignment of the vertex buffer stride.
     pub min_vertex_input_binding_stride_alignment: buffer::Offset,
+}
 
+/// Resource limits related to Descriptor Indexing.
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct DescriptorIndexingCapabilities {
```

Comment:
This is a direct mapping from `vkPhysicalDeviceDescriptorIndexingPropertiesEXT`, which is probably more than what we want for `gfx`.